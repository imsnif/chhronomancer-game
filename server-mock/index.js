import playerStore from '../stores/player'
import timelineStore from '../stores/timeline'
import powerStore from '../stores/power'
import statsStore from '../stores/stats'

const timelines = [
  {name: 'Timeline 1', type: 'steal', players: []},
  {name: 'Timeline 2', type: 'assist', players: []},
  {name: 'Timeline 3', type: 'prevent', players: []},
  {name: 'Timeline 4', type: 'reset', players: []}
]

const players = [
  {
    id: 1,
    name: 'Aram',
    items: [
      {name: 'assist', source: false},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: false},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_me_s.png')
  },
  {
    id: 2,
    name: 'Tom',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 1'},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_tom_s.png')
  },
  {
    id: 3,
    name: 'Gondollieri',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 3'},
      {name: 'unlock', source: 'Timeline 1'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_gondollieri_s.png')
  },
  {
    id: 4,
    name: 'Simmons',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: false},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 2'},
      {name: 'unlock', source: 'Timeline 2'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_simmons_s.png')
  }
]

const powers = [
  {
    playerId: 1,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 1,
    timelineName: 'Timeline 2',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 400000,
    target: {
      type: 'timeline',
      name: 'Timeline 2'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 2,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 1000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 2, score: 1},
      {id: 1, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 2}
    ]
  },
  {
    playerId: 3,
    timelineName: 'Timeline 1',
    name: 'Combining',
    startTime: Date.now(),
    endTime: Date.now() + 2000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 3, score: 1},
      {id: 1, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 4, score: 5}
    ]
  },
  {
    playerId: 4,
    timelineName: 'Timeline 1',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 4, score: 7},
      {id: 2, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 1, score: 5}
    ]
  }
]

export default function mockServer () {
  timelines.forEach((t) => timelineStore.addTimeline(t))
  players.forEach(p => {
    playerStore.addPlayer(p)
  })
  statsStore.changePlayerId(1)
  timelineStore.addPlayer('Timeline 1', 1)
  timelineStore.addPlayer('Timeline 1', 2)
  timelineStore.addPlayer('Timeline 2', 2)
  timelineStore.addPlayer('Timeline 3', 2)
  timelineStore.addPlayer('Timeline 1', 3)
  timelineStore.addPlayer('Timeline 2', 3)
  timelineStore.addPlayer('Timeline 3', 3)
  timelineStore.addPlayer('Timeline 1', 4)
  timelineStore.addPlayer('Timeline 3', 4)
  powers.forEach(p => powerStore.addPower(p))
  timelineStore.changeTimelineLock('Timeline 1', true)
}
