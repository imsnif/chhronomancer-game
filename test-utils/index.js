module.exports = {
  mockFB () {
    const login = require('react-native-facebook-login')
    const FBLogin = jest.fn()
    FBLogin.mockReturnValue(null)
    login.FBLogin = FBLogin
    login.FBLoginManager = {LoginBehaviors: {Web: 2}}
    return FBLogin
  },
  navigatorMock (routeName) {
    return {
      getCurrentRoutes: () => [routeName],
      push: () => {}
    }
  }
}
