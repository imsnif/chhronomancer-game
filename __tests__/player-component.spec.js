import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { navigatorMock } from '../test-utils'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

beforeEach(() => {
  jest.useFakeTimers()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('Player component => renders correctly', () => {
  const timelineName = 'Timeline 1'
  const playerId = 1
  const Player = require('../components/player').default
  require('../server-mock') // TODO: export a function like a normal Human being
  const tree = renderer.create(
    <Player timelineName={timelineName} id={playerId} navigator={navigatorMock()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Player component => renders correctly without active power', () => {
  const timelineName = 'Timeline 2'
  const playerId = 2
  const Player = require('../components/player').default
  require('../server-mock') // TODO: export a function like a normal Human being
  const tree = renderer.create(
    <Player timelineName={timelineName} id={playerId} navigator={navigatorMock()} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
