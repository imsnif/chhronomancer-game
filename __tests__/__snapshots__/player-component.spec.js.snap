// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component => renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#329932",
        "borderColor": "#329932",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginBottom": 7,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
          }
        }
        style={
          Object {
            "backgroundColor": "#141313",
            "borderColor": "black",
            "borderWidth": 1,
            "flex": 0,
            "height": 28,
            "margin": 1,
            "resizeMode": "stretch",
            "width": 26,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "green",
              "height": "100%",
              "opacity": 0.5,
              "width": "100%",
            }
          }
        />
      </Image>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#0e0e0e",
            "flex": 7,
            "fontFamily": "telegrama_raw",
            "textAlign": "center",
          }
        }
      >
        Aram
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "#141313",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 13,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "paddingTop": 25,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "fontFamily": "telegrama_raw",
                "fontSize": 6,
                "marginLeft": 9,
                "paddingTop": 2,
                "textAlign": "left",
              }
            }
          >
            Resetting
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 7,
                "fontFamily": "telegrama_raw",
                "fontSize": 6,
                "paddingTop": 2,
                "textAlign": "left",
              }
            }
          >
            | |==------------------------------------------------|   1% 00:01:39
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 16,
                "marginRight": 4,
                "width": 16,
              }
            }
          />
          <Image
            source={1}
            style={
              Object {
                "height": 16,
                "marginRight": 4,
                "width": 16,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "flexDirection": "row",
                "fontFamily": "telegrama_raw",
                "fontSize": 8,
                "marginTop": 10,
              }
            }
          >
            Artifacts: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "flexDirection": "row",
                "fontFamily": "telegrama_raw",
                "fontSize": 8,
                "marginTop": 10,
              }
            }
          >
            8
            /7
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Player component => renders correctly without active power 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#329932",
        "borderColor": "#329932",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginBottom": 7,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
          }
        }
        style={
          Object {
            "backgroundColor": "#141313",
            "borderColor": "black",
            "borderWidth": 1,
            "flex": 0,
            "height": 28,
            "margin": 1,
            "resizeMode": "stretch",
            "width": 26,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "green",
              "height": "100%",
              "opacity": 0.5,
              "width": "100%",
            }
          }
        />
      </Image>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#0e0e0e",
            "flex": 7,
            "fontFamily": "telegrama_raw",
            "textAlign": "center",
          }
        }
      >
        Tom
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "#141313",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 13,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "paddingTop": 25,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 16,
                  "marginRight": 4,
                  "width": 16,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 16,
                "marginRight": 4,
                "width": 16,
              }
            }
          />
          <Image
            source={1}
            style={
              Object {
                "height": 16,
                "marginRight": 4,
                "width": 16,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "flexDirection": "row",
                "fontFamily": "telegrama_raw",
                "fontSize": 8,
                "marginTop": 10,
              }
            }
          >
            Artifacts: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "flexDirection": "row",
                "fontFamily": "telegrama_raw",
                "fontSize": 8,
                "marginTop": 10,
              }
            }
          >
            6
            /7
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
